pkg/scalers/couchdb_scaler.go

```diff
@@ -0,0 +1,247 @@
+		}
+		fmt.Println(res)
+	}
+	fmt.Println("Count of rows: ", count)
+	return count, nil
+}
+
+func (s *couchDBScaler) IsActive(ctx context.Context) (bool, error) {
+	result, err := s.getQueryResult(ctx)
+	if err != nil {
+		s.logger.Error(err, fmt.Sprintf("failed to get query result by couchDB, because of %v", err))
+		return false, err
+	}
+	return result > s.metadata.activationQueryValue, nil
+}
+
+func parseCouchDBMetadata(config *ScalerConfig) (*couchDBMetadata, string, error) {
+	var connStr string
+	var err error
+	// setting default metadata
+	meta := couchDBMetadata{}
+
+	// parse metaData from ScaledJob config
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, "", fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %v", queryValue, err.Error())
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, "", fmt.Errorf("no queryValue given")
+	}
+
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %v", activationQueryValue, err.Error())
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	meta.dbName, err = GetFromAuthOrMeta(config, "dbName")
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Resolve connectionString
+	switch {
+	case config.AuthParams["connectionString"] != "":
+		meta.connectionString = config.AuthParams["connectionString"]
+	case config.TriggerMetadata["connectionStringFromEnv"] != "":
+		meta.connectionString = config.ResolvedEnv[config.TriggerMetadata["connectionStringFromEnv"]]
+	default:
+		meta.connectionString = ""
+		meta.host, err = GetFromAuthOrMeta(config, "host")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.port, err = GetFromAuthOrMeta(config, "port")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.username, err = GetFromAuthOrMeta(config, "username")
+		if err != nil {
+			return nil, "", err
+		}
+
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+		}
+		if len(meta.password) == 0 {
+			return nil, "", fmt.Errorf("no password given")
+		}
+	}
+
+	if meta.connectionString != "" {
+		connStr = meta.connectionString
+	} else {
+		// Build connection str
+		addr := fmt.Sprintf("%s:%s", meta.host, meta.port)
+		// auth := fmt.Sprintf("%s:%s", meta.username, meta.password)
+		connStr = "http://" + addr
+	}
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("couchdb-%s", val))
+	}
```

Comment:
We don't support the metric name modification as it's an internal part. Old scalers support it due to legacy reasons, but new scalers shouldn't do it. The metric name is an internal thing that shouldn't be exposed if it's explicitly required by the scaler behaviour

Reply:
What should I replace with it then?

Reply:
Just remove it and generate it the similar way it is done for other scalers, for example here: https://github.com/kedacore/keda/blob/d3ed916db06ef441e172e61b2ea995fff65025ff/pkg/scalers/activemq_scaler.go#L162